packages/openid4vc/src/openid4vc-issuer/router/redirectEndpoint.ts

```diff
@@ -0,0 +1,190 @@
+              authorizationServerMetadata,
+              pkceCodeVerifier: issuanceSession.chainedIdentity.pkceCodeVerifier,
+
+              redirectUri: joinUriParts(config.baseUrl, [issuer.issuerId, 'redirect']),
+            })
+            .catch((error) => {
+              // TODO: improve the thrown error by looking into the specific error.
+              // if (error instanceof Oauth2ClientErrorResponseError)
+
+              throw new Oauth2ServerErrorResponseError(
+                {
+                  error: Oauth2ErrorCodes.ServerError,
+                  error_description: 'Error processing authorization code',
+                },
+                {
+                  internalMessage: `Error occurred during retrieval of access token from ${authorizationServerMetadata.issuer}.`,
+                  cause: error,
+                }
+              )
+            })
+
+          // Grant authorization
+          const kms = agentContext.resolve(Kms.KeyManagementApi)
+          const authorizationCode = TypedArrayEncoder.toBase64URL(kms.randomBytes({ length: 32 }))
+          const authorizationCodeExpiresAt = addSecondsToDate(new Date(), config.authorizationCodeExpiresInSeconds)
+
+          const redirectUri = new URL(issuanceSession.chainedIdentity.redirectUri)
+          redirectUri.searchParams.set('code', authorizationCode)
+
+          if (issuanceSession.chainedIdentity.state) {
+            redirectUri.searchParams.set('state', issuanceSession.chainedIdentity.state)
+          }
+
+          // Update authorization information
+          issuanceSession.authorization = {
+            ...issuanceSession.authorization,
+            code: authorizationCode,
+            codeExpiresAt: authorizationCodeExpiresAt,
+          }
+
+          // Store access token response
+          issuanceSession.chainedIdentity = {
+            ...issuanceSession.chainedIdentity,
+            externalAccessTokenResponse: accessTokenResponse,
+          }
+
+          // TODO: we need to start using locks so we can't get corrupted state
+          await openId4VcIssuerService.updateState(
+            agentContext,
+            issuanceSession,
+            OpenId4VcIssuanceSessionState.AuthorizationGranted
+          )
+
+          return response.redirect(redirectUri.toString())
+        }
+
+        if (error && typeof error === 'string') {
+          issuanceSession.errorMessage = `Error occurred during identity chaining. ${error}: ${request.query.error_description}`
+          await openId4VcIssuerService.updateState(agentContext, issuanceSession, OpenId4VcIssuanceSessionState.Error)
+
+          const redirectUri = new URL(issuanceSession.chainedIdentity.redirectUri)
+          redirectUri.searchParams.set('error', Oauth2ErrorCodes.ServerError)
+
+          // TODO: correctly map the error codes from the underlying oauth2 flow
+          // redirectUri.searchParams.set('error_description')
```

Comment:
I think it would be good to add support for the most common params. e.g `access_denied` and `invalid_grant`

Reply:
Why don't we just pass forward the full error? Is there a reason not to do it? If we only want some errors, how do we decide where to put the border between what we want to forward and what we don't?